tsdb/engine/tsm1/engine.go

```diff
@@ -1947,9 +1937,13 @@ func (e *Engine) WriteSnapshot() (err error) {
 }
 
```

Comment:
I'd avoid using a `bool` value here to control this behavior.  Obviously passing `bool` is fine if you want to convey that value to something like:

```
func SetUserAdmin(b bool)
```

where we're passing it because we actually need the `bool` as a value.  This is a case, I think, it only exists to change behavior.  That usually signals to me that it is time for a new function.

Or, in this case, do you think it'd be possible to just call `e.WriteSnapshot()` instead of `CreateSnapshot()`?

Reply:
I believe this review thread should be on the `CreateSnapshot()` method that was modified below and not on the `WriteSnapshot()` method.

I think avoiding `bool` parameters is good practice in general because some functions tend to grow. However, `CreateSnapshot()` has remained simple and mostly unchanged over time. It takes only two steps: 1) snapshot in-mem write cache to a file on disk then 2) take a snapshot of all the files on disk. I think it would be overkill to refactor this into separate functions to avoid the `bool` parameter.

Reply:
Per @dgnorton , will probably leave this as a boolean, but agree on best practices with you.